dace/codegen/targets/rtl.py

```diff
@@ -338,13 +490,68 @@ def unparse_tasklet(self, sdfg: sdfg.SDFG, dfg: state.StateSubgraphView,
         tasklet = node
 
         # construct variables paths
-        unique_name: str = "top_{}_{}_{}".format(sdfg.sdfg_id,
-                                                 sdfg.node_id(state),
-                                                 state.node_id(tasklet))
+        unique_name: str = "{}_{}_{}_{}".format(tasklet.name, sdfg.sdfg_id,
+                                                sdfg.node_id(state),
+                                                state.node_id(tasklet))
+
+        # Collect all of the input and output connectors into buses and scalars
+        buses = {}
+        scalars = {}
+        for edge in state.in_edges(tasklet):
+            arr = sdfg.arrays[edge.src.data]
+            # catch symbolic (compile time variables)
+            check_issymbolic([
+                tasklet.in_connectors[edge.dst_conn].veclen,
+                tasklet.in_connectors[edge.dst_conn].bytes
+            ], sdfg)
+
+            # extract parameters
+            vec_len = int(
+                symbolic.evaluate(tasklet.in_connectors[edge.dst_conn].veclen,
+                                  sdfg.constants))
+            total_size = int(
+                symbolic.evaluate(tasklet.in_connectors[edge.dst_conn].bytes,
+                                  sdfg.constants))
+            if isinstance(arr, data.Array):
+                if self.mode == 'xilinx':
```

Comment:
`mode = 'xilinx'`? That doesn't sound like a mode -- what is difference between this field and the existing combination of vendor + mode?

Reply:
Fixed in https://github.com/carljohnsen/dace/commit/4a7e567b636a6801f476fc97d8e6b7d74acde946